manifests/repo/community.pp

```diff
@@ -0,0 +1,50 @@
+# @summary Private class to manage sensu community repository resources
+# @api private
+#
+class sensu::repo::community {
+
+  case $facts['os']['family'] {
+    'RedHat': {
+      if $facts['os']['name'] == 'Amazon' {
+        if $facts['os']['release']['major'] =~ /^201\d$/ {
+          $repo_release = '6'
+        } else {
+          $repo_release = '7'
+        }
+      } else {
+        $repo_release = $facts['os']['release']['major']
+      }
+      yumrepo { 'sensu_community':
+        ensure          => 'present',
+        baseurl         => "https://packagecloud.io/sensu/community/el/${repo_release}/\$basearch",
+        descr           => 'sensu_community',
+        enabled         => 1,
+        gpgcheck        => 0,
+        gpgkey          => 'https://packagecloud.io/sensu/community/gpgkey',
+        metadata_expire => 300,
+        repo_gpgcheck   => 1,
+        sslcacert       => '/etc/pki/tls/certs/ca-bundle.crt',
+        sslverify       => 1,
+}
+    }
+    'Debian': {
+      apt::source { 'sensu_community':
+        ensure   => 'present',
+        location => "https://packagecloud.io/sensu/community/${downcase($facts['os']['name'])}/",
+        repos    => 'main',
+        release  => $facts['os']['distro']['codename'],
+        include  => {
+          'src' => true,
+        },
+        key      => {
+          'id'     => '7F54E8A5C0CB51DBE612D2F50156BD72FEC8CD59',
+          'source' => 'https://packagecloud.io/sensu/community/gpgkey',
+        },
+      }
+    }
+    default: {
```

Comment:
perhaps the default should be to emit a `warning()`

Reply:
It's not possible to ever hit this default, it's just there for lint.  The `sensu` class will fail if `osfamily` is not `RedHat` or `Debian`.

Reply:
That's what should happen :)

Reply:
To avoid the empty case statement, suggest doing two if statements

```puppet
if $facts['os']['family'] == 'RedHat' {
  ...
}

if $facts['os']['family'] == 'Debian' {
  ...
}
```

This is also easier to read than a if-elsif

Reply:
Implemented.  I'll open separate pull request to do this in `sensu::repo`.